cdap-sentry/cdap-sentry-extension/cdap-sentry-model/src/main/java/co/cask/cdap/security/authorization/sentry/model/Artifact.java

```diff
@@ -75,7 +81,14 @@ public AuthorizableType getAuthzType() {
    */
   @Override
   public String getName() {
-    return artifactName + ARTIFACT_DETAILS_SEPARATOR + artifactVersion;
+    return artifactVersion == null ? artifactName : artifactName + ARTIFACT_DETAILS_SEPARATOR + artifactVersion;
+  }
+
+  @Nullable
+  @Override
+  public String getSubType() {
+    // Artifact does not have a sub type
+    return null;
```

Comment:
Version is not considered as sub type. The only sub-type we have now is program type.